src/YesSql.Provider.Common/BaseDialect.cs

```diff
@@ -89,6 +89,8 @@ public virtual string GetDropForeignKeyConstraintString(string name)
         }
 
         public virtual bool SupportsIfExistsBeforeTableName => false;
+        public virtual bool SupportsIfExistsBeforeIndexName => true;
```

Comment:
I would assume that the `GetDropIndexString` is all you need, just update it for each dialect that has a different behavior than the most common way.

Reply:
All 4 dialects are different in one way or another.
Shall I just make an abstract `GetDropIndexString` method and implement for each dialect? 

Just trying to follow the existing patterns...

Reply:
Yes, I would prefer a single method in this case, an no custom flags. I don't remember the history of the other flag you took as an example. Doesn't look like something I would have done ;)

Reply:
Cool, yeah I couldn't figure them out for ages either